packages/NodeTypeResolver/NodeTypeResolver.php

```diff
@@ -69,6 +76,7 @@ public function __construct(
         private readonly AccessoryNonEmptyStringTypeCorrector $accessoryNonEmptyStringTypeCorrector,
         private readonly IdentifierTypeResolver $identifierTypeResolver,
         private readonly RenamedClassesDataCollector $renamedClassesDataCollector,
+        private readonly ExprPrinter $exprPrinter,
```

Comment:
I could not find the container defintion of rector. wondered that it does not include all phpstan services by default which phpstan defines

https://github.com/phpstan/phpstan-src/blob/0745fbd69a92d1dc164992601a1f28548f096d42/conf/config.neon#L535

any hints on where I can define this required service would be awesome.

Reply:
You can define phpstan service in `config/phpstan/*` neon files under services:

https://github.com/rectorphp/rector-src/blob/main/config/phpstan/static-reflection.neon

Reply:
@staabm You have to register factory services in Rector config here:

https://github.com/rectorphp/rector-src/blob/34e1cf13fdd613cb732d082cfcfa86bbdb7db5dc/config/config.php#L210-L211

And add the factory bridge to PHPStan container here:

https://github.com/rectorphp/rector-src/blob/34e1cf13fdd613cb732d082cfcfa86bbdb7db5dc/packages/NodeTypeResolver/DependencyInjection/PHPStanServicesFactory.php#L93-L99

Also, copy-pasting 5 lines from the `ExprPrinter` might be easier solution, as it's really just printing of code. Depends what you find easier to mimic :+1:

Reply:
thanks. added the services but it still doesn't work. do I need to manually re-dump the container or invalidate a cache somewhere?

> Symfony\Component\DependencyInjection\Exception\RuntimeException: Cannot autowire service "Rector\NodeTypeResolver\NodeTypeResolver": argument "$exprPrinter" of method "__construct()" references class "PHPStan\Node\Printer\ExprPrinter" but no such service exists.

Reply:
Add to `config/config` like what @TomasVotruba suggestion above should make it work

Reply:
will look at it later.. I am afk for a few hours. if anyone of you guys is willing to add the necessary configs, that would be awesome

Reply:
I'll check it :+1:  Thanks and enjoy free time